azure-ts-webserver-component/webserver.ts

```diff
@@ -0,0 +1,111 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure";
+import { specialArchiveSig } from "@pulumi/pulumi/runtime";
+
+// All resources will share a resource group.
+let resourceGroupName = new azure.core.ResourceGroup("server").name;
+
+// Create a network and subnet for all VMs.
+let network = new azure.network.VirtualNetwork("server-network", {
+    resourceGroupName,
+    addressSpaces: [ "10.0.0.0/16" ],
+    subnets: [{
+        name: "default",
+        addressPrefix: "10.0.1.0/24",
+    }],
+});
+let subnet = new azure.network.Subnet("server-subnet", {
+    resourceGroupName,
+    virtualNetworkName: network.name,
+    addressPrefix: "10.0.2.0/24",
+});
+
+/**
+ * WebServer is a reusable web server component that creates and exports a NIC, public IP, and VM.
+ */
+export class WebServer extends pulumi.ComponentResource {
+    public readonly publicIp: azure.network.PublicIp;
+    public readonly networkInterface: azure.network.NetworkInterface;
+    public readonly vm: azure.compute.VirtualMachine;
+
+    /**
+     * Allocate a new web server VM, NIC, and public IP address.
+     * @param name The name of the web server resource.
+     * @param args A bag of arguments to control the web server VM creation.
+     */
+    constructor(name: string, args: WebServerArgs) {
+        super("ws-ts-azure-comp:webserver:WebServer", name);
```

Comment:
seems unfortunate to not takes 'opts' here, just to be consistent with the Component pattern.  But it is an example... so maybe it matters less?